e2e_playwright/st_plotly_chart_select_test.py

```diff
@@ -0,0 +1,182 @@
+    )
+    assert_snapshot(chart, name="st_plotly_chart-single_select")
+
+    app.keyboard.down("Shift")
+    app.mouse.move(445, 375)
+    app.mouse.down()
+    app.mouse.up()
+    wait_for_app_run(app, wait_delay=3000)
+    assert_snapshot(chart, name="st_plotly_chart-double_select")
+    assert_snapshot(
+        app.get_by_test_id("stDataFrame"), name="st_plotly_chart-double_select_df"
+    )
+
+    chart.scroll_into_view_if_needed()
+    chart.hover()
+    app.mouse.dblclick(400, 400)
+    wait_for_app_run(app, 3000)
+
+    expect(app.get_by_test_id("stDataFrame")).to_have_count(0)
+    assert_snapshot(chart, name="st_plotly_chart-bar_chart_reset")
+
+
+def test_box_select_on_stacked_bar_chart_displays_a_df(app: Page):
+    chart = app.locator(".stPlotlyChart").nth(3)
+    chart.scroll_into_view_if_needed()
+    expect(chart).to_be_visible()
+    chart.hover()
+    app.mouse.down()
+    app.mouse.move(50, 50)
+    app.mouse.down()
+    app.mouse.move(150, 150)
+    app.mouse.up()
+    wait_for_app_run(app, 3000)
+    expect(app.get_by_test_id("stDataFrame")).to_have_count(1)
+
+
+# TODO(willhuang1997): Readd choropleth charts and add a working test
+# This test could be flakey because https://github.com/plotly/plotly.js/issues/6898
+# def test_click_on_choroleth_chart_displays_a_df(app: Page):
+#     chart = app.locator(".stPlotlyChart").nth(4)
+#     chart.scroll_into_view_if_needed()
+#     expect(chart).to_be_visible()
+#     chart.hover()
+#     app.mouse.down()
+#     app.mouse.up()
+#     expect(app.get_by_test_id("stDataFrame")).to_have_count(1)
+
+
+# TODO(willhuang1997): Looks like webkit is not working but it is working locally
+@pytest.mark.only_browser("chromium")
+def test_lasso_select_on_histogram_chart_displays_a_df_and_resets_when_double_clicked(
+    app: Page, assert_snapshot: ImageCompareFunction
+):
+    chart = app.locator(".stPlotlyChart").nth(4)
+    chart.scroll_into_view_if_needed()
+    expect(chart).to_be_visible()
+    chart.hover()
+    app.mouse.down()
+    app.mouse.move(350, 350)
+    app.mouse.move(375, 375)
+    app.mouse.move(400, 400)
+    app.mouse.move(435, 500)
+    app.mouse.up()
+    wait_for_app_run(app, 3000)
+    expect(app.get_by_test_id("stDataFrame")).to_have_count(1)
+
+    chart = app.locator(".stPlotlyChart").nth(4)
+    chart.scroll_into_view_if_needed()
+
+    app.mouse.dblclick(400, 400)
+    chart = app.locator(".stPlotlyChart").nth(4)
+    chart.scroll_into_view_if_needed()
+    wait_for_app_run(app, 3000)
+    assert_snapshot(chart, name="st_plotly_chart-reset")
+
+
+def test_double_click_select_mode_doesnt_reset_zoom(
+    app: Page, assert_snapshot: ImageCompareFunction
+):
+    chart = app.locator(".stPlotlyChart").nth(0)
+    expect(chart).to_be_visible()
+    chart.hover()
+    app.mouse.down()
+    app.mouse.move(50, 50)
+    app.mouse.down()
+    app.mouse.move(150, 150)
+    app.mouse.up()
+    wait_for_app_run(app, 3000)
```

Comment:
nit: I don't think we need to wait an additional 3 seconds here since the next operation is a waiting except operation